vpr/src/route/rr_node.cpp

```diff
@@ -11,10 +11,9 @@ short t_rr_node::length() const {
 }
 
 bool t_rr_node::edge_is_configurable(t_edge_size iedge) const {
-    auto iswitch = edge_switch(iedge);
-
     auto& device_ctx = g_vpr_ctx.device();
-
+    const auto& rr_graph = device_ctx.rr_graph;
```

Comment:
Can you double check the dependencies among ``rr_node.cpp`` and ``rr_graph_storage.cpp`` and ``rr_node_impl.h``?>
My understanding, ``edge_switch()`` is a method of ``t_rr_node``. We do not need to use the RRGraphView API here.

Reply:
`edge_switch()` is a method of `t_rr_node` which is being overridden  in `rr_node_impl.h`. We are introducing the APIs in `rr_graph_view.h` to delete the functions of `rr_node_impl.h`. Once #1917 is merged, we will move `edge_is_configurable()` and `validate()` from `rr_node.cpp` to `rr_graph_view.h` as we are trying to shadow the `rr_node.cpp` by introducing APIs to `rr_graph_view.h`

Reply:
Make sense. I am fine with this plan.

Reply:
Please leave a comment
```
/* TODO: This API should be reworked once rr_switch APIs are in RRGraphView. */
```